flake.nix

```diff
@@ -0,0 +1,36 @@
+{
+  description = "Devshell flake";
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    rust-overlay = {
+      inputs = {
+        flake-utils.follows = "flake-utils";
+        nixpkgs.follows = "nixpkgs";
+      };
+      url = "github:oxalica/rust-overlay";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, rust-overlay } @ inputs:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit overlays system;
+        };
+      in
+      {
+        formatter = pkgs.nixpkgs-fmt;
+
+        devShells.default = pkgs.mkShell {
+          buildInputs = with pkgs; [
+            go
+            gofumpt
+            golangci-lint
+          ];
```

Comment:
The rust overlay is in here, but isn't rust missing in the build inputs?

Reply:
Yeah, it is. I kinda forgot about it because I usually install Rust via `rustup` on my Nix machines regardless (kind of a bad habit but yeah).

I'll add it.